src/NUnitFramework/tests/Constraints/CollectionEquivalentConstraintResultTests.cs

```diff
@@ -0,0 +1,125 @@
+using System;
+using System.Collections.Generic;
+
+namespace NUnit.Framework.Constraints
+{
+    [TestFixture]
+    public class CollectionEquivalentConstraintResultTests
+    {
+        private List<string> _expected = new List<string>() { "one", "two" };
+
+        private CollectionEquivalentConstraint _constraint;
+        private TextMessageWriter _writer;
+
+        [SetUp]
+        public void TestSetup()
+        {
+            _constraint = new CollectionEquivalentConstraint(_expected);
+            _writer = new TextMessageWriter();
+        }
+
+        [Test]
+        public void TestOneMissingValue()
+        {
+            List<object> actualList = new List<object>() { "one" };
+
+            ConstraintResult cr = _constraint.ApplyTo(actualList);
+            cr.WriteMessageTo(_writer);
+
+            string expectedMsg = 
+                "  Expected: equivalent to < \"one\", \"two\" >" + Environment.NewLine +
+                "  But was:  < \"one\" >" + Environment.NewLine +
+                "  Missing (1): < \"two\" >" + Environment.NewLine;
+            Assert.AreEqual(expectedMsg, _writer.ToString());
+        }
+
+        [Test]
+        public void TestTwoMissingValues()
+        {
+            List<object> actualList = new List<object>() { };
+
+            ConstraintResult cr = _constraint.ApplyTo(actualList);
+            cr.WriteMessageTo(_writer);
+
+            string expectedMsg = 
+                "  Expected: equivalent to < \"one\", \"two\" >" + Environment.NewLine +
+                "  But was:  <empty>" + Environment.NewLine +
+                "  Missing (2): < \"one\", \"two\" >" + Environment.NewLine;
+            Assert.AreEqual(expectedMsg, _writer.ToString());
+        }
+
+        [Test]
+        public void TestOneExtraValue()
+        {
+            List<object> actualList = new List<object>() { "three", "one", "two" };
+
+            ConstraintResult cr = _constraint.ApplyTo(actualList);
+            cr.WriteMessageTo(_writer);
+
+            string expectedMsg = 
+                "  Expected: equivalent to < \"one\", \"two\" >" + Environment.NewLine +
+                "  But was:  < \"three\", \"one\", \"two\" >" + Environment.NewLine +
+                "  Extra (1): < \"three\" >" + Environment.NewLine;
+            Assert.AreEqual(expectedMsg, _writer.ToString());
+        }
+
+        [Test]
+        public void TestOneExtraOneMissing()
+        {
+            List<object> actualList = new List<object>() { "three", "one", };
+
+            ConstraintResult cr = _constraint.ApplyTo(actualList);
+            cr.WriteMessageTo(_writer);
+
+            string expectedMsg = 
+                "  Expected: equivalent to < \"one\", \"two\" >" + Environment.NewLine +
+                "  But was:  < \"three\", \"one\" >" + Environment.NewLine +
+                "  Missing (1): < \"two\" >" + Environment.NewLine +
+                "  Extra (1): < \"three\" >" + Environment.NewLine;
```

Comment:
Although perhaps there's a less verbose way of displaying both "Extra" and "Missing" separately?